receiver/pluginreceiver/plugin_test.go

```diff
@@ -492,3 +494,44 @@ func TestCheckParameters(t *testing.T) {
 		})
 	}
 }
+
+func Test_checkExtensions(t *testing.T) {
+	tmpDir := t.TempDir()
+	testCases := []struct {
+		name        string
+		extenstions map[string]any
+		pluginName  string
+		expectedErr error
+	}{
+		{
+			name:        "Valid Extensions",
+			extenstions: map[string]any{"file_storage": map[string]any{"directory": tmpDir}},
+			pluginName:  "plugin_one",
+		},
+		{
+			name:        "Invalid Extensions Decoding",
+			extenstions: map[string]any{"file_storage": "hello"},
+			expectedErr: errors.New("'' expected a map, got 'string'"),
+		},
+		{
+			name:        "Invalid Extensions Directory",
+			extenstions: map[string]any{"file_storage": map[string]any{"directory": "/invalid/directory"}},
+			expectedErr: errors.New(""),
+		},
```

Comment:
This is failing on Windows; I think this is actually a valid directory, it just requires root permissions on mac/linux.

Not sure what do use here as an invalid directory, I'd be OK with just removing this test case.